### PR TITLE
Make the height of the editor in Review Box smaller (4 lines as GitHub)

### LINKED ISSUE #18183: Review form overflows in height
### Gitea Version

https://github.com/go-gitea/gitea/commit/d228d346c4d719cc0c0725ff1409eb97f0ac3109

### Operating System

Arch Linux

### Browser Version

Firefox 97.0a1 

### Can you reproduce the bug on the Gitea demo site?

Yes

### Description

See: https://try.gitea.io/Gusted/Test-repo/pulls/1/files -> Click on Review button.

I'm not comfortable enough with CSS to fix this one myself, nor sure how it should be "wrapped". 

### Screenshots

image

### PR DESCRIPTION
**Pull request by wxiaoguang**
As the title (and see the comment in PR)

This PR tries to improve the UI. And partially helps
* https://github.com/go-gitea/gitea/issues/18183

# Before

image

# After

## PR Review

image

## Issue Edit

image

### CODE REVIEW

**Comment by zeripath** [comment]
can we shrink the size of the dropzone too?

**Review by zeripath — APPROVED** [review]

**Comment by wxiaoguang** [comment]
> can we shrink the size of the dropzone too?

I have tried but found it a little more difficult. Since we are working for 1.16-rc, time is limited, maybe we can refactor the Dropzone later (and with others together).

**Comment by silverwind** [comment]
Should probably remove the dropzone completely and make the edit box act as a drop target instead. But I agree it's something for another PR.

File: web_src/js/features/comp/EasyMDE.js [hunk 1]
```diff
@@ -61,7 +62,7 @@ export async function createCommentEasyMDE(textarea) {
   }
 
   const EasyMDE = await importEasyMDE();
-  const easyMDE = new EasyMDE({
+  const easyMDE = new EasyMDE(Object.assign({
```
Comment (silverwind) [inline web_src/js/features/comp/EasyMDE.js]: Nit: Maybe use spread syntax instead? `{...{opts}, ...easyMDEOptions}`

**Comment by zeripath** [comment]
wouldn't this simply target the dropzone to make it smaller?

```css
.review-box .dz-message {
   margin-top: 10px;
   margin-bottom: 10px;
}
```

**Comment by wxiaoguang** [comment]
> OK that's giving a slightly odd appearance due to the min-height rules but we could stick a slightly higher margin-top on the .dz-message - or even do
> 
> ```css
> .review-box .ui.dropzone.dz-clickable {
>    min-height: 0!important;
> }
> ```

image


But if we drop something it, it will be much larger 😂

I think we could use some other design instead of introducing too many `!important`.

**Comment by zeripath** [comment]
The fact that it grows when it actually contains something is fine by me!

We don't need it to take up all that space when it's empty and is likely to be empty for the majority of the time.

**Comment by zeripath** [comment]
(unfortunately the `!important` is necesitated because of another `!important` elsewhere. I agree `!important` is AWFUL but as you correctly say - this is a bodge to fix things for 1.16.)

**Comment by wxiaoguang** [comment]
The `!important` is not necessary. I can handle it.

How do you think about the two new UI?

image


image

File: web_src/js/features/comp/EasyMDE.js (hunk 1 above)
Comment (wxiaoguang) [reply web_src/js/features/comp/EasyMDE.js]: Done

**Comment by zeripath** [comment]
now that looks a bit tight around the drop files to me! 

(Sorry this is going round and round slightly.)

**Comment by wxiaoguang** [comment]
> now that looks a bit tight around the drop files to me!
> 
> (Sorry this is going round and round slightly.)

That's fine. I also feel the height is a little small. Let's make it larger.

image



image

**Comment by zeripath** [comment]
And if we get rid of the margin-bottom from the editor `<div class="field">` or put the dropzone in the same `<div class="field">` that should make them more connected...

**Comment by wxiaoguang** [comment]
image

**Review by lunny — APPROVED** [review]

**Comment by wxiaoguang** [comment]
> How about move the drop box on top of lines and words counting and move lines and words counting on left bottom corner.

It's impossible now. `lines and words counting` is inside the EasyMDE, it is a part of it.

**Comment by wxiaoguang** [comment]
trigger bot

**Pull request merged**